### PR TITLE
Frontend: US12 - Non-Interactive Mode #35

### PR DESCRIPTION
**Pull request by Asiya-Ismail**
# Non Interactive Mode Feature:

## Description:
- This feature has a function that translates the graph to make sure latest nodes of the graph are always shown.
- The function was made in G6Graph file.
-  A store file was also created to store the interactive mode variable to be used between G6Graph and App.svelte components.
- This feature also has a display message function which displays a message letting the user know of the time the last received node was pushed to the graph
- This feature was implemented in the App.svelte where we used a time interval to check for when we stop receiving nodes.
- The message with the time stamp is displayed on the screen after a minute of not receiving any new nodes. 

### Display Message with Timestamp:
Display Message

## Testing:
- Testing was done using the current event generator.

### CODE REVIEW

**Comment by RuthgerD** [comment]
please rebase properly so that unrelated commits are not included

**Comment by Asiya-Ismail** [comment]
Is this what you meant?

**Comment by RuthgerD** [comment]
This works :+1:

File: frontend/src/App.svelte [hunk 1]
```diff
@@ -68,18 +64,70 @@
...
+  let time:Date = new Date();
+  if ( time.getDate() == dayLastEventRecieved){
+      dayToDisplay = "TODAY"; 
+      
+  }else if (time.getDate() - dayLastEventRecieved == 1){
+    
+      dayToDisplay = "YESTERDAY"; 
+  }else if (time.getDate() - dayLastEventRecieved> 1){
+      dayToDisplay = displayDate;
+  }
```
Comment (gusaxtcha) [inline frontend/src/App.svelte:88]: Nice detail of Today and yesterday! Makes it really neat!

File: frontend/src/App.svelte [hunk 2]
```diff
@@ -68,18 +64,70 @@
...
+}
+
+  let ms = 60000;
+ let interval= setInterval( displayInfoMessage, ms);
+
+ const startTimer = () =>{
+  clearInterval(interval);
+  interval= setInterval( displayInfoMessage, ms);
+}
+
```
Comment (gusaxtcha) [inline frontend/src/App.svelte:107]: All nice and neat, I would just add sufficient comments.

File: frontend/src/components/G6Graph.svelte [hunk 3]
```diff
@@ -3,42 +3,62 @@
...
+            let scrollDistanceWithRatio:number = 0;
+            let zoomRatio:number = graph.getZoom(); 
+            nodePoint = e.x - container.scrollWidth;
+            if(nodePoint != oldNodePoint){ // find out if node has moved to new position
+              scrollDistance = nodePoint - oldNodePoint; 
+              scrollDistanceWithRatio = scrollDistance*zoomRatio // to recalculate translation based on zoom ratio
+              graph.translate(-scrollDistanceWithRatio,0);
+            }
+        }
+    }
```
Comment (gusaxtcha) [inline frontend/src/components/G6Graph.svelte:59]: This is all nice as well but I would add more comments, it makes it easier to read!

File: frontend/src/components/G6Graph.svelte [hunk 4]
```diff
@@ -67,196 +84,194 @@
...
-            data: timeBarData,
-            smooth: true,
-          },
-          tick: {
-            tickLabelFormatter: (timeBarData: any) => {
-              return "";
-            },
-
-            tickLineStyle: {
-              fill: "#f28c18",
```
Comment (gusaxtcha) [inline frontend/src/components/G6Graph.svelte:103]: Not what the reason for moving so much code around was 😅 It makes it really difficult to find the code written for this feature.

**Review by gusaxtcha — COMMENTED** [review]
Overall I think this is really useful and displayed nicely. As per my comments, I would suggest adding more comments and avoid moving code around as much so it's easy to see what has been added. Good work!

File: frontend/src/components/G6Graph.svelte (hunk 3 above)
Comment (Asiya-Ismail) [reply frontend/src/components/G6Graph.svelte:59]: Understood, will do!

**Comment by Asiya-Ismail** [comment]
@gusaxtcha Added new comments. Let me know what you think.

**Pull request merged**